service/resource/persistentvolume/update.go

```diff
@@ -34,13 +35,66 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState inter
 		return nil
 	}
 
-	_, err = toPV(obj)
+	pv, err := toPV(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	// update state logic
+	switch combinedState := string(rpv.State) + rpv.RecycleState; combinedState {
+	case "ReleasedRecycled":
+		pv, err := r.newRecycleStateAnnotation(pv, released)
+		_, err = r.k8sClient.Core().PersistentVolumes().Update(pv)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	case "ReleasedReleased":
+		err = r.k8sClient.Core().PersistentVolumes().Delete(pv.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	case "AvailableCleaning":
+		pvcdef := newPvc(pv)
+		pvc, err := r.k8sClient.Core().PersistentVolumeClaims("kube-system").Create(pvcdef)
+		if err != nil {
+			return microerror.Maskf(err, "failed to create persistent volume claim", pvc.Name)
+		}
+	case "BoundCleaning":
+		pvcName := fmt.Sprintf("pv-cleaner-claim-%s", pv.Name)
+		pvc, err := r.k8sClient.Core().PersistentVolumeClaims("kube-system").Get(pvcName, metav1.GetOptions{})
+		if err != nil {
+			return microerror.Maskf(err, "failed to get persistent volume claim", pvcName)
+		}
+	
+		cleanupJobDef := newCleanupJob(pvc)
+		cleanupJob, err := r.k8sClient.Batch().Jobs("kube-system").Create(cleanupJobDef)
+		if err != nil {
+			cleanupJob, err = r.k8sClient.Batch().Jobs("kube-system").Get(cleanupJobDef.GetName(), metav1.GetOptions{})
+			if err != nil {
+				return microerror.Maskf(err, "Failed to create pvc", pvc.GetName())
+			}
```

Comment:
You should check IsAlreadyExist error matcher instead: https://github.com/giantswarm/crdstorage/blob/54b678b94c166ee23e333c3b9408bb61552fc77e/storage.go#L120